map2loop/thickness_calculator.py

```diff
@@ -340,3 +347,208 @@ def compute(
+            basal_contacts (geopandas.GeoDataFrame): basal contact geo data with locations and unit names of
+            the contacts (columns must contain ["ID","basal_unit","type","geometry"])
+            structure_data (pandas.DataFrame): sampled structural data
+            map_data (map2loop.MapData): a catchall so that access to all map data is available
+
+        Returns:
+        --------
+            pandas.DataFrame: units dataframe with added thickness columns:
+            "gammaThickness" is the median thickness of the unit,
+            "gammaStdDev" is the standard deviation of the thickness of the unit
+
+        Note:
+        -----
+            This method is highly dependent on the existence of stratigraphic measurements that follow the strike of the geological contacts.
+            If an unit does not contain a stratigraphic measurement, the thickness will not be calculated. Future versions of map2loop will attempt to solve for this.
+            If the thickness is not calculated for a given unit, the assigned thickness will be -1.
+            For the bottom and top units of the stratigraphic sequence, the assigned thickness will also be -1.
+        """
+        sampled_contacts = map_data.sampled_contacts
+        sampled_structures = structure_data
+        basal_contacts = basal_contacts.copy()
+
+        geology = map_data.get_map_data(datatype=Datatype.GEOLOGY)
+        geology[['minx', 'miny', 'maxx', 'maxy']] = geology.bounds
+
+        sampled_structures = geopandas.GeoDataFrame(
+            sampled_structures,
+            geometry=geopandas.points_from_xy(sampled_structures.X, sampled_structures.Y),
+            crs=basal_contacts.crs,
+        )
+        sampled_structures['unit_name'] = geopandas.sjoin(sampled_structures, geology)['UNITNAME']
+        sampled_basal_contacts = rebuild_sampled_basal_contacts(basal_contacts, sampled_contacts)
+
+        basal_contacts_bu = basal_contacts.copy()
+        basal_contacts_bu['geometry'] = basal_contacts_bu.buffer(0.1)
+
+        map_dx = geology.total_bounds[2] - geology.total_bounds[0]
+        map_dy = geology.total_bounds[3] - geology.total_bounds[1]
+
+        ths = []
+        lis = []
+
+        for s in range(0, len(sampled_structures)):
+            th = None
+            measurement = sampled_structures.iloc[s]
+            measurement_pt = shapely.Point(measurement.X, measurement.Y)
+            litho_in = measurement['unit_name']
+            strike = (measurement['DIPDIR'] - 90) % 360
+
+            bbox_poly = geology[geology['UNITNAME'] == litho_in][['minx', 'miny', 'maxx', 'maxy']]
+
+            GEO_SUB = geology[geology['UNITNAME'] == litho_in]['geometry'].values[0]
+            neighbour_list = list(
+                basal_contacts[GEO_SUB.intersects(basal_contacts.geometry)]['basal_unit']
+            )
+            B = calculate_endpoints(measurement_pt, strike, 10000, bbox_poly)
```

Comment:
line length should be an attribute of the calculator, so that it can be changed by the user.

Reply:
@lachlangrose could you please confirm if this change is enough? - https://github.com/Loop3D/map2loop/pull/59/commits/23aeb06c0f422598e054727ab0d8357da76a7705

Reply:
I will resolve this for now, but happy to re-address this @ 3.1 review if necessary.